controllers/front/ajax.php

```diff
@@ -46,8 +46,20 @@ public function postProcess()
                     );
                 }
 
-                $paymentFee = new Number(Tools::getValue('paymentFee'));
+                $paymentFee = trim($paymentFee);
                 $orderTotal = new Number((string)$cart->getOrderTotal());
+
+                if (strpos($paymentFee, '%') !== false) {
+                    $paymentFee = str_replace('%', '', $paymentFee);
+                    $paymentFee = new Number((string)$paymentFee);
+                    $percentage = $paymentFee->dividedBy(new Number('100'));
+                    $paymentFee = $orderTotal->times($percentage);
+                } else if ($paymentFee > 0) {
+                    // The fee is a flat amount.
+                    $paymentFee = new Number((string)$paymentFee);
+                }
+                $buckarooFee = Tools::displayPrice($paymentFee->toPrecision(2));
```

Comment:
these functionality is used in another class, maybe extract this in a separate service class